### PR TITLE
Fix URL parsing

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: <link>
N/A

### Overview

Cast AnyURL to str so that it can be split

### Rationale

N/A

### Juju Events Changes

N/A

### Module Changes

charm.py

### Library Changes

N/A

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -93,7 +93,9 @@ def reconcile(self) -> None:
         )
         if self.state.custom_config_repository:
             wazuh.configure_git(
-                container, self.state.custom_config_repository, self.state.custom_config_ssh_key
+                container,
+                str(self.state.custom_config_repository),
```
Comment (yhaliaw) [inline src/charm.py:97]: Where is the str splitting performed?
Is the `configure_git` a method from a library that expects a str as input?

Comment (arturo-seijas) [reply src/charm.py:97]: Inside that method, which is in another module

**Review by alithethird — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (yhaliaw) [reply src/charm.py:97]: Ok, that method is taking a str, so the str casting should be a good idea.

**Review by yhaliaw — APPROVED** [review]

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def test_reconcile_reaches_active_status_when_repository_configured(
     wazuh_install_certificates_mock.assert_called_with(container, ANY, "somecert")
     wazuh_update_configuration_mock.assert_called_with(container, ["10.0.0.1"])
     configure_git_mock.assert_called_with(
-        container, wazuh_config.custom_config_repository, "somekey"
+        container, str(wazuh_config.custom_config_repository), "somekey"
```
Comment (Thanhphan1147) [inline tests/unit/test_charm.py:71]: Do we actually need to cast it to a string or do we just need a typing.cast for mypy?

**Review by Thanhphan1147 — APPROVED** [review]
LGTM, just a minor comment

File: tests/unit/test_charm.py (hunk 2 above)
Comment (arturo-seijas) [reply tests/unit/test_charm.py:71]: We need to cast it, it crashes otherwise

**Pull request merged**